Autopilot/Src/MPXV7002DP.cpp

```diff
@@ -0,0 +1,91 @@
+#include "airspeed.hpp"
+#include "adc.h"
+#include <stddef.h>
+
+#include <cmath>
+
+#define VS 5
+#define DENSITY_AIR 1.2754
+#define ADC_RESOLUTION 4096
+#define KPA_TO_PA 1000
+#define BASELINE 2048 // the real baseline should be 2047.5, but this is close enough
+
+MPXV7002DP* MPXV7002DP::s_Instance = nullptr;
+
+MPXV7002DP::MPXV7002DP() {
+    //constructor
+    HAL_StatusTypeDef status = HAL_ADC_Start(&hadc3);
+    if(status != HAL_OK) {
+        sensorStatus = 1;
+    }
+
+    CalcOffset();
+}
+
+MPXV7002DP* MPXV7002DP::GetInstance() {
+    if(!s_Instance) {
+        s_Instance = new MPXV7002DP;
+    }
+    return s_Instance;
+}
+
+uint16_t MPXV7002DP::GetAverage() {
+	uint16_t avg = 0;
+	for(int i = 0; i < 10; i++) {
+		HAL_ADC_PollForConversion(&hadc3, 10);
```

Comment:
nice, so 10 ms was the lowest time you could get?

Reply:
That was the lowest I went to, Ill try to get it lower.